pdf/internal/e2etest/validate.go

```diff
@@ -0,0 +1,70 @@
+/*
+ * This file is subject to the terms and conditions defined in
+ * file 'LICENSE.md', which is part of this source code package.
+ */
+
+package e2etest
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/unidoc/unidoc/common"
+)
+
+// To enable ghostscript validation, the path to the binary needs to be specified.
+// Set environment variable:
+//		UNIDOC_GS_BIN_PATH to the path of the ghosccript binary (gs).
+var (
+	ghostscriptBinPath = os.Getenv("UNIDOC_GS_BIN_PATH")
+)
+
+// validatePdf a pdf file using Ghostscript, returns an error if unable to execute.
+// Also returns the number of output warnings, which can be used as some sort of measure
+// of validity, especially when comparing with a transformed version of same file.
+func validatePdf(path string, password string) (int, error) {
+	if len(ghostscriptBinPath) == 0 {
+		return 0, errors.New("UNIDOC_GS_BIN_PATH not set")
+	}
+	common.Log.Debug("Validating: %s", path)
+
+	var cmd *exec.Cmd
```

Comment:
This is somewhat subjective because I don't like repeating code in if/else statements but this could also be written as:

```
	params := []string{"-dBATCH", "-dNODISPLAY", "-dNOPAUSE"}
	if len(password) > 0 {
		params = append(params, fmt.Sprintf("-sPDFPassword=%s", password))
	}
	params = append(params, path)

	var out bytes.Buffer
	var errOut bytes.Buffer

	cmd := exec.Command(ghostscriptBinPath, params...)
	cmd.Stdout = &out
	cmd.Stderr = &errOut
```